src/SFML/Window/Context.cpp

```diff
@@ -49,19 +49,40 @@ namespace sf
 Context::Context() : m_context(priv::GlContext::create())
 {
     if (!setActive(true))
-    {
         err() << "Failed to set context as active during construction" << std::endl;
-    }
 }
 
 
 ////////////////////////////////////////////////////////////
 Context::~Context()
 {
-    if (!setActive(false))
-    {
+    if (m_context && !setActive(false))
         err() << "Failed to set context as inactive during destruction" << std::endl;
-    }
+}
+
+
+////////////////////////////////////////////////////////////
+Context::Context(Context&& other) noexcept : m_context(std::move(other.m_context))
+{
+    if (!setActive(true))
+        err() << "Failed to set context as active during construction" << std::endl;
```

Comment:
Are you sure this is wanted? When constructing from a context that is not active, wouldn't it make more sense to get a context that is also not active? What about this:
```suggestion
    if (ContextImpl::currentContext == &other)
        ContextImpl::currentContext = this;
```

Remark also applies to move assignment operator.

---

I tried it locally. It became apparent to me that the logic about `currentContext` in `setActive` looks wrong: `setActive(false)` sets `currentContext` to `nullptr` even when the context being deactivated is not the active one.
This implementation of `setActive` looks more correct to me:
```c++
bool Context::setActive(bool active)
{
    const bool result = m_context->setActive(active);

    if (result)
    {
        if (active)
            ContextImpl::currentContext = this;
        else if (ContextImpl::currentContext == this)
            ContextImpl::currentContext = nullptr;
    }

    return result;
}
```

Reply:
> It became apparent to me that the logic about currentContext in setActive looks wrong: setActive(false) sets currentContext to nullptr even when the context being deactivated is not the active one.

I can reproduce this with a test. Create a 2nd context then call `.setActive(false)` on the first context and then `sf::Context::getActiveContext` returns false. Good catch.

I think I want to make a separate PR to address this.

Reply:
See #2816